pulser-core/pulser/register/special_layouts.py

```diff
@@ -25,27 +25,38 @@
 if TYPE_CHECKING:
     from pulser.register import Register
 
+import numpy as np
+from scipy.spatial.distance import cdist
 
-class SquareLatticeLayout(RegisterLayout):
-    """A RegisterLayout with a square lattice pattern in a rectangular shape.
+
+class RectangularLatticeLayout(RegisterLayout):
+    """A RegisterLayout with a rectangular lattice pattern in a rectangular
+    shape.
 
     Args:
         rows: The number of rows of traps.
         columns: The number of columns of traps.
-        spacing: The distance between neighbouring traps (in µm).
+        x_spacing: Horizontal distance between neighbouring traps (in µm).
+        y_spacing: Vertical distance between neighbouring traps (in µm)
     """
 
-    def __init__(self, rows: int, columns: int, spacing: float):
-        """Initializes a SquareLatticeLayout."""
+    def __init__(
+        self, rows: int, columns: int, x_spacing: float, y_spacing: float
+    ):
+        """Initializes a RectangularLatticeLayout."""
         self._rows = int(rows)
         self._columns = int(columns)
-        self._spacing = float(spacing)
+        self._x_spacing = float(x_spacing)
+        self._y_spacing = float(y_spacing)
         slug = (
-            f"SquareLatticeLayout({self._rows}x{self._columns}, "
-            f"{self._spacing}µm)"
+            f"RectangularLatticeLayout(Shape : {self._rows}x{self._columns}, "
+            f"Lattice pattern : {self._x_spacing}x{self._y_spacing}µm)"
```

Comment:
This slug is a bit too long, how about just this:
```suggestion
            f"RectangularLatticeLayout({self._rows}x{self._columns},  "
            f"{self._row_spacing}x{self._col_spacing}µm)"
```